### PR TITLE
Allow to pass file ID/path as initial input text field value

### PR DESCRIPTION
**Pull request by julien-nc**
This will be helpful to make file actions (in the Files app).

The `OCA.Assistant.openAssistantForm` function now accepts inputs like 
``` js
{ input: { fileId: 123 } }
```
or
``` js
{ input: { filePath: '/path/to/file.txt' } }
```
and the parsed file content will be inserted in the input field on the fly.

### CODE REVIEW

File: src/components/AssistantTextProcessingForm.vue [hunk 1]
```diff
@@ -315,23 +315,53 @@ export default {
...
+				if (taskType.inputShape[k]?.type === 'Text') {
+					if (this.myInputs[k]?.fileId || this.myInputs[k]?.filePath) {
+						const { filePath, fileId } = { fileId: this.myInputs[k]?.fileId, filePath: this.myInputs[k]?.filePath }
+						this.myInputs[k] = ''
+						this.parseFile({ fileId, filePath })
+							.then(response => {
+								if (response.data?.ocs?.data?.parsedText) {
+									this.myInputs[k] = response.data?.ocs?.data?.parsedText
+								}
+							})
```
Comment (kyteinsky) [inline src/components/AssistantTextProcessingForm.vue:333]: an empty catch block would be good here

**Comment by kyteinsky** [comment]
let's update the docs with an example too (https://github.com/nextcloud/assistant/blob/main/docs/developer/web-integration.md)

File: src/components/AssistantTextProcessingForm.vue (hunk 1 above)
Comment (marcelklehr) [reply src/components/AssistantTextProcessingForm.vue:333]: Could even send a toast that it failed

Comment (julien-nc) [reply src/components/AssistantTextProcessingForm.vue:334]: Done

File: docs/developer/web-integration.md [hunk 2]
```diff
@@ -47,15 +46,17 @@ The promise resolves with a task object which looks like:
 ```javascript
 {
 	appId: 'text',
-	category: 1, // 0: text generation, 1: image generation, 2: speech-to-text
 	id: 310, // the assistant task ID
-	identifier: 'my custom identifier',
-	inputs: { prompt: 'give me a short summary of a simple settings section about GitHub' },
+	customId: 'my custom identifier',
+	input: { input: 'give me a short summary of a simple settings section about GitHub' },
```
Comment (kyteinsky) [inline docs/developer/web-integration.md:51]: `inputs` first key here, typo

Comment (julien-nc) [reply docs/developer/web-integration.md:51]: Done, thanks

File: docs/developer/web-integration.md [hunk 3]
```diff
@@ -47,15 +46,17 @@ The promise resolves with a task object which looks like:
...
 	ocpTaskId: 152, // the underlying OCP task ID
-	output: 'blabla',
-	status: 3, // 0: unknown, 1: scheduled, 2: running, 3: sucessful, 4: failed
-	taskType: 'OCP\\TextProcessing\\FreePromptTaskType',
-	timestamp: 1711545305,
+	output: { output: 'blabla' },
```
Comment (kyteinsky) [inline docs/developer/web-integration.md:53]: and outputs perhaps? I didn't see the code.

**Review by kyteinsky — APPROVED** [review]
nice :tada: 
and neat feature!

**Pull request merged**